pkg/cache/cluster.go

```diff
@@ -167,6 +217,67 @@ type clusterCache struct {
 	eventHandlers               map[uint64]OnEventHandler
 }
 
+func (c *clusterCache) StoreApisMeta(key schema.GroupKind, value *apiMeta) {
+	c.apisMeta.Store(key, value)
+}
+
+func (c *clusterCache) LoadApisMeta(key schema.GroupKind) (*apiMeta, error) {
+	v, ok := c.apisMeta.Load(key)
+	if !ok {
+		return nil, fmt.Errorf("not found key: %s in apiMeta map", key.String())
+	}
+
+	meta, ok := v.(*apiMeta)
+	if !ok {
+		return nil, fmt.Errorf("stored type is invalid in apiMeta map")
+	}
+
+	if meta == nil {
+		return nil, fmt.Errorf("stored type is nil in apiMeta map")
+	}
+
+	return meta, nil
+}
+
+func (c *clusterCache) DeleteApisMeta(key schema.GroupKind) {
+	c.apisMeta.Delete(key)
+}
+
+func (c *clusterCache) ApisMetaLength() int {
+	var cnt int
+	c.apisMeta.Range(func(key, value interface{}) bool {
+		cnt++
+		return true
+	})
+	return cnt
+}
+
+func (c *clusterCache) StoreNsIndex(key string, value *Resources) {
+	c.nsIndex.Store(key, value)
+}
+
+func (c *clusterCache) LoadNsIndex(key string) (*Resources, error) {
```

Comment:
Same as above - panicking should be ok if something unexpected is in this private field.